### PR TITLE
Add return types

### PR DESCRIPTION
**Pull request by Crell**
Follow on to #23

### CODE REVIEW

**Review by ashnazg — APPROVED** [review]

File: src/CacheItemInterface.php [hunk 1]
```diff
@@ -46,7 +46,7 @@ public function getKey();
      * @return mixed
      *   The value corresponding to this cache item's key, or null if not found.
      */
-    public function get();
+    public function get(): mixed;
```
Comment (Nyholm) [inline src/CacheItemInterface.php:49]: Sorry, for my ignorance, but does `mixed` include null?

**Review by Nyholm — APPROVED** [review]

File: src/CacheItemInterface.php (hunk 1 above)
Comment (ashnazg) [reply src/CacheItemInterface.php:49]: Yes, `null` is included:  https://www.php.net/manual/en/language.types.declarations.php#language.types.declarations.mixed

File: src/CacheItemInterface.php [hunk 2]
```diff
@@ -72,12 +72,12 @@ public function isHit();
      * @return static
      *   The invoked object.
      */
-    public function set($value);
+    public function set(mixed $value): static;
```
Comment (samdark) [inline src/CacheItemInterface.php:75]: Does that mean that immutability is required?

File: src/CacheItemInterface.php [hunk 3]
```diff
@@ -72,12 +72,12 @@ public function isHit();
...
 
     /**
      * Sets the expiration time for this cache item.
      *
-     * @param \DateTimeInterface|null $expiration
+     * @param ?\DateTimeInterface $expiration
```
Comment (samdark) [inline src/CacheItemInterface.php]: I think it should stay as is: https://docs.phpdoc.org/3.0/guide/references/phpdoc/types.html

Comment (Crell) [reply src/CacheItemInterface.php]: Doesn't that make it inconsistent with the native types, then?

(PHPDoc's own spec is well over a decade old at this point, no?  That's the reason PSR-5 and friends, er, tried to exist...)

File: src/CacheItemInterface.php (hunk 2 above)
Comment (Crell) [reply src/CacheItemInterface.php:75]: `CacheItem` explicitly cannot be an immutable object.  It's a builder object.  set() must return "the invoked object", so it's going to be of the same type (static) no matter what.  If it's not, it's an error.

File: src/CacheItemInterface.php (hunk 3 above)
Comment (samdark) [reply src/CacheItemInterface.php]: No. It's not inconsistent with PHP 8 union types and it's current PHPDocumentor 3 syntax that has wide support among IDEs and static code analysis tools.

File: src/CacheItemPoolInterface.php [hunk 4]
```diff
@@ -89,7 +89,7 @@ public function clear();
      * @return bool
      *   True if the item was successfully removed. False if there was an error.
      */
-    public function deleteItem($key);
+    public function deleteItem(string $key): bool;
```
Comment (goetas) [inline src/CacheItemPoolInterface.php:92]: would it make sense to allow also the new `Stringable` type (`string|\Stringable`)? (same question for the other places where `string` is used as parameter)

Comment (Crell) [reply src/CacheItemPoolInterface.php:92]: It's a judgement call either way.  I went with `string|Stringable` only when the existing documentation implied an object was OK.  In this case, the key should be a trivial opaque string, not a formatted string like a URL or IP address (where __toString makes sense to use).  I cannot think of when using an object as the key ID would make sense, and even if someone is, tossing `(string)` into the function call is trivial for them to do.

Comment (goetas) [reply src/CacheItemPoolInterface.php:92]: I've used the `__toString` in situations where computing the key is expensive, so `Stringable` was allowing me to have sort of "lazy strings", but thinking it twice, in this context `string` is good.

**Comment by nicolas-grekas** [comment]
This just broke projects that tests with the dev-version.
Please update the branch-alias asap.

**Pull request merged**